### ISSUE TITLE
Reproduce paper accuracy

### ISSUE THREAD

**Issue by chaoji1013** [zjunlp/MKG_Analogy#9]
参照之前问题，我已在原论文更改如下代码。这里也有个问题，为何更改这个代码后，提升精度那么大？

#features['sep_idx'] = torch.tensor(sep_idx)
features['attention_mask'] = features['attention_mask'].unsqueeze(1).expand([features['input_ids'].size(0), features['input_ids'].size(1), features['input_ids'].size(1)]).clone()
for i, idx in enumerate(sep_idx):
    features['attention_mask'][i, :idx[2], idx[2]:] = 0

按照说明运行mkgformer.sh文件，这里以hits@10为指标评价，展示以下结果：

1.自己预训练MarKG，结果为0.71，论文附录的表9为0.779；基于这个预训练再去微调MARS，结果为0.331，论文表2为0.408.
2.运行checkpoint "pretrain_mkgformer.ckpt"，结果为0.779，与论文相同；基于"pretrain_mkgformer.ckpt"微调MARS，结果为0.402.
3.直接测试checkpoint "ft_mart_mkgformer.ckpt"，结果为0.408，与论文相同。

所以有一些疑问，请您麻烦解答一下，非常感谢。
对于结果1，自己按照给定的sh文件预训练MarKG，为何仅为0.71.
对于结果2，在checkpoint "pretrain_mkgformer.ckpt"微调MARS，结果0.402，随机种子设置是一样的，但还是和论文的0.408差一点。

**Comment by flow3rdown** [comment]
1. 这段代码相当于直接设定`g_ea=0` 和`g_ae=1`，对于MKGformer帮助比较大，但也可以通过学习这两个参数来逼近或达到更优结果。
2. 不同环境可能表现不同，建议你可以试下调整下学习率和alpha等参数，学习率可设置在e-5级别。
3. 0.402相比0.408差距已经不大，所以可能还是因为超参的原因，可以试下调一下学习率等超参。

**Comment by chaoji1013** [comment]
您好，抱歉再次打扰，我之前是一直按照源代码中给的学习率5e-5去跑的，听取了您的建议我在尝试从学习率1e-5到9e-5依次去跑。我使用的也是单卡3090，环境感觉没太大差别，但是在第一阶段的预训练上精度0.71与论文的0.779却差别有些大。

我在想是否训练数据集我这边不对称，或者代码有些出入。我这边是直接下载的源代码，下载后也没动过，数据集我直接在https://www.zjukg.org/project/MKG_Analogy/ 下载，MarKG有3307个三元组，感觉也没差。唯一改动了一下图像数据集，因为有的图像名字太长导致读取失败，我把有的图像名字缩短了一下。

**Comment by flow3rdown** [comment]
感谢您的关注哈，我会再跑一下确认下结果，可能会晚点给您回复

**Comment by chaoji1013** [comment]
您好，请问大概什么时候能够反馈一下最新的情况，目前我尝试第一阶段训练的结果还是和原论文差别较大。

**Comment by zxlzr** [comment]
您好，最近计算资源太紧张，我们仍在通过训练排查问题，也可能是硬件gpu不一样和不同精度训练等导致的差异

**Comment by flow3rdown** [comment]
> 您好，请问大概什么时候能够反馈一下最新的情况，目前我尝试第一阶段训练的结果还是和原论文差别较大。

您好，我们排查到了问题并更新了代码，您可以拉取最新代码并尝试一下第一阶段训练的结果

**Comment by zxlzr** [comment]
您好，后续还有什么问题吗，不知道您的issue是否已解决

**Comment by chaoji1013** [comment]
您好，我拉取了最新代码，我看是去掉了词嵌入参数的冻结。我立马尝试了第一阶段的训练，结果和原论文结果一样，解决了之前的问题。但是当我再次尝试第二阶段的微调时，发现结果和原论文差别还是挺大，希望能麻烦再看一下这个第二阶段微调是否有问题。下面展示了第二阶段微调的结果：
||hits1|hits3|hits5|hits10|mrr|
| :---: | :---: | :---: | :---: | :---: | :---: |
论文结果|0.301|0.367|0.380|0.408|0.341|
|复现结果|0.302|0.348| 0.367|0.386|0.332|

**Comment by zxlzr** [comment]
您好，您的结果已经在正常的方差范围内了，建议您多跑几次取均值，随机种子和不同的GPU环境对结果存在一定影响的。

**Comment by chaoji1013** [comment]
好吧，那我多尝试几次，感谢。

**Comment by flow3rdown** [comment]
hit@1的结果基本正常了，也可以试下不同图片的影响，可以在`DataCollatorForSeq2Seq`中随机选择图片
